mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +61,115 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
                 .setVersion(releaseInfoJson.get("tag_name").textValue())
                 .setDescription(basicInfoJson.get("description").textValue())
                 .setServiceCrawlingOrigin(MicoServiceCrawlingOrigin.GITHUB)
                 .setGitCloneUrl(basicInfoJson.get("clone_url").textValue())
                 .setGitReleaseInfoUrl(releaseInfoJson.get("url").textValue());
+            if (!StringUtils.isEmpty(dockerfilePath)) {
+                if (existsFileInGithubRepo(gitHubRepositoryApiUrl, dockerfilePath)) {
+                    micoService.setDockerfilePath(dockerfilePath);
+                } else {
+                    throw new IllegalArgumentException("The Dockerfile path must be a valid path relativ to the repository root");
+                }
+            }
+            return micoService;
+        } else {
+            throw new IllegalArgumentException("An error occurred while requesting information about the GitHub repository. Send two requests and got the status codes "
+                + responseBasicInfo.getStatusCode() + " and " + responseReleaseInfo.getStatusCode().is2xxSuccessful());
+        }
 
-        } catch (IOException e) {
-            log.error(e.getStackTrace().toString());
-            log.error("Getting exception '{}'", e.getMessage());
-            throw e;
+    }
+
+    /**
+     * Checks if a file in a GitHub repository exists
+     *
+     * @param uriBasicInfo   the URI to a specific release of a GitHub repository
+     * @param dockerfilePath the relative path to the Dockerfile. The path is relative to the root directory of the
+     *                       GitHub repository
+     * @return {@code true} if the Dockerfile exists in the in the specified repository. {@code false} if response status is HTTP NOT FOUND.
+     * @throws {@link HttpClientErrorException} in case of HTTP 4XX response from GitHub
+     * @throws {@link HttpStatusCodeException} in case of HTTP 5XX response from GitHub
+     */
+    private boolean existsFileInGithubRepo(String uriBasicInfo, String dockerfilePath) {
+        ResponseEntity<String> responseDockerFileInfo;
+        UriComponentsBuilder dockerFileUriBuilder = UriComponentsBuilder.fromHttpUrl(uriBasicInfo);
+        UriComponents dockerFileUriComponent = dockerFileUriBuilder.pathSegment(GITHUB_API_CONTENTS).pathSegment(dockerfilePath).build();
+        log.debug("Check if the Dockerfile exists at {}", dockerFileUriComponent.toString());
+        try {
+            responseDockerFileInfo = restTemplate.getForEntity(dockerFileUriComponent.toUri(), String.class);
+            HttpStatus responseStatus = responseDockerFileInfo.getStatusCode();
+            if (responseStatus.equals(HttpStatus.OK)) {
+                log.debug("The file {} exists", dockerfilePath);
+                return true;
+            }
+        } catch (HttpClientErrorException.NotFound e) {
+            log.info("The file{} does not exist", dockerfilePath);
```

Comment:
Missing space between "file" and argument

Reply:
Done